core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -64,8 +64,7 @@
     private int queryManagerExecutorPoolSize = 5;
 
     private Duration remoteTaskMaxErrorDuration = new Duration(5, TimeUnit.MINUTES);
-    private int remoteTaskMaxCallbackThreads = 1000;
-
+    private int remoteTaskMaxCallbackThreads = Runtime.getRuntime().availableProcessors() * 2;
```

Comment:
1.I would go with `task.concurrency` * 2. The default result will be the same, but it seems more reasonable to follow concurrency than actual CPU count.
2.Is there any real chance that there is going to be a lot of IO-bound threads and this will become a bottleneck?

Reply:
This should really be done similar as
```
private int taskConcurrency = nextPowerOfTwo(getAvailablePhysicalProcessorCount());
```